exercises/dot-dsl/example.py

```diff
@@ -0,0 +1,52 @@
+NODE, EDGE, ATTR = range(3)
+
+
+class Node(object):
+    def __init__(self, name, attrs={}):
+        self.name = name
+        self.attrs = attrs
+
+    def __eq__(self, other):
+        return self.name == other.name and self.attrs == other.attrs
+
+
+class Edge(object):
+    def __init__(self, src, dst, attrs={}):
+        self.src = src
+        self.dst = dst
+        self.attrs = attrs
+
+    def __eq__(self, other):
+        return (self.src == other.src and
+                self.dst == other.dst and
+                self.attrs == other.attrs)
+
+
+class Graph(object):
+    def __init__(self, data=[]):
+        self.nodes = []
+        self.edges = []
+        self.attrs = {}
+
+        if not isinstance(data, list):
+            raise TypeError("Graph data malformed")
+
+        for item in data:
+            if len(item) != 3 and len(item) != 4:
+                raise TypeError("Graph item malformed")
```

Comment:
Is this check really necessary? Surely the conditionals below will achieve the same but with a more informative output?

Reply:
hmm but without these checks, 

```
            type_ = item[0]
```

might raise an `IndexError` if `len(item) < 1`.

Or should I just say

```
            if len(item) < 3:
                raise TypeError("Graph item incomplete")
```

?

Reply:
Well spotted, I think that would be a sensible change.

Reply:
Done!